nerfstudio/configs/annotated_types.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 the Regents of the University of California, Nerfstudio Team and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+from collections import OrderedDict
+from typing import TYPE_CHECKING
+
+import tyro
+
+from nerfstudio.configs.dataparser_configs import dataparser_configs
+from nerfstudio.configs.external_methods import get_external_methods
+from nerfstudio.configs.method_configs import descriptions as method_descriptions
+from nerfstudio.configs.method_configs import method_configs
+from nerfstudio.data.dataparsers.base_dataparser import DataParserConfig
+from nerfstudio.engine.trainer import TrainerConfig
+from nerfstudio.plugins.registry import discover_dataparsers, discover_methods
+
+
+def merge_methods(methods, method_descriptions, new_methods, new_descriptions, overwrite=True):
```

Comment:
while we're making changes, can we add annotations here?

Reply:
Sure